Marlin/src/sd/SdBaseFile.cpp

```diff
@@ -703,14 +703,18 @@ bool SdBaseFile::open(SdBaseFile *dirFile, const uint8_t dname[11]
               // Get VFat dir entry
               pvFat = (vfat_t *) p;
               // Get checksum from the last entry of the sequence
-              if (pvFat->sequenceNumber & 0x40) lfnChecksum = pvFat->checksum;
+              if (pvFat->sequenceNumber & 0x40) {
+                lfnChecksum = pvFat->checksum;
+                memset(lfnName, '\0', sizeof(lfnName));
```

Comment:
Do we need to clear the whole buffer here to ensure a null terminator later, or can we get away with just `lnfName[0] = '\0'`?

Reply:
No, I guess it wouldn't work: `lfnName` is filled from the end of the actual LFN found in VFAT sequences (which appear in reverse order; see method `getLFNName`), which means just using `lfnName[0] = '\0'` wouldn't suffice.
However, the code below 
```cpp
if (!strncasecmp((char*)dlname, (char*)lfnName, lfnNameLength)) lfnFileFound = true;
```
compares these chararrays up until `lfnNameLength`, which is initialized as
```cpp
lfnNameLength = useLFN ? strlen((char*)dlname) : 0,
```
above, which means we should compare these strings up until requested part length. Which means, yes, we probably don't even need this line at all and it should work OK for comparison.

Reply:
Generally, when copying strings into buffers we should append a nul after writing the other bytes. This avoids the need to clear the whole buffer before the copy. The `memset` might use some extra electrons, but it's minor thing in practical terms. Only the code size ultimately matters, not the performance, since this is not a bottleneck.